MyApplication/app/src/main/java/com/joelimyx/myapplication/MainActivity.java

```diff
@@ -0,0 +1,124 @@
+package com.joelimyx.myapplication;
+
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.Log;
+import android.view.View;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+import android.widget.Button;
+import android.widget.EditText;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.FormBody;
+import okhttp3.Headers;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+public class MainActivity extends AppCompatActivity {
+    public static final String ENCODED = "V1RZd0k3Y0N6cE43VmNCYVVIT0dSb252OTpzNEM3aG9tT2FxSkpFeXV1ZWJNNHlGQXd2ZkFid0xqN21uNThMMjhKQmVGUzU0VzdPSQ==";
+    private static String mToken;
+
+    RecyclerView mRecyclerView;
+    EditText mEditText;
+    Button mButton;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        mRecyclerView = (RecyclerView) findViewById(R.id.recyclerview);
+        mEditText = (EditText) findViewById(R.id.input);
+        mButton = (Button) findViewById(R.id.button);
+        mRecyclerView.setLayoutManager(new LinearLayoutManager(this,LinearLayoutManager.VERTICAL,false));
+        getAccessToken();
+        View.OnClickListener listener = new View.OnClickListener() {
```

Comment:
Might be helpful to add some comments or at least blank lines so your code is more readable